x-pack/legacy/plugins/index_management/public/app/components/mappings_editor/reducer.ts

```diff
@@ -57,12 +65,20 @@ export interface State {
     format(): MappingsFields;
     isValid: boolean;
   };
+  templates: {
+    defaultValue: {
+      dynamic_templates: MappingsTemplates['dynamic_templates'];
+    };
+    form?: FormHook<MappingsTemplates>;
```

Comment:
I made a change on my current PR in the logic of the configuration form to be more specific instead of passing the whole `form`. You can see it here: https://github.com/elastic/kibana/pull/54241/files#diff-c7f4f4960afc300ef126e2c1a3d857b8R48

Can we make the same change here?

It also means to be more specific when we dispatch the update, as you can see here: https://github.com/elastic/kibana/pull/54241/files#diff-dfcd0122417fdb550ad2c83268216bd6R96

As a bonus, @cjcenizal will love you for that! 😊

Reply:
I'm happy to make this change, however, I see it will also require some of the changes you made to `mappings_editor.tsx`. I think it might make more sense to wait to adopt this until after your PR is merged.